components/studentPortal/freemium/FreemiumUnitNodeView.tsx

```diff
@@ -0,0 +1,129 @@
+import React from "react";
+import { Button } from "../../ui/Button";
+
+export type FreemiumUnitNodeViewProps = {
+  title: string;
+  description: string;
+  completed: boolean;
+  locked: boolean;
+  hiddenLine: boolean;
+  type: "lesson" | "quiz" | "assignment";
+  grayedOut: boolean;
+  freemiumMessage: boolean;
+};
+
+export const FreemiumUnitNodeView: React.FC<FreemiumUnitNodeViewProps> = ({
+  completed,
+  locked,
+  hiddenLine,
+  title,
+  description,
+  type,
+  grayedOut,
+  freemiumMessage,
+}: FreemiumUnitNodeViewProps) => {
+  const imageSrc = (completed, locked, type, freemiumMessage) => {
```

Comment:
Could we extract this UI logic into a shared component that can be used by both the freemium and default unit view component.

Maybe called NodeIcon

Reply:
good idea.  put the logic in the api folder and then created a NodeIcon component used by both freemium and regular students.